Apps/Scigraph/dwim/draw.lisp

```diff
@@ -34,110 +34,40 @@ advised of the possiblity of such damages.
-	#+clim-0.9 :flipping-ink
-	#+clim-1.0 clim:+flipping-ink+
-	#+clim-2 clim:+flipping-ink+
-	#-clim :flip)
-(defvar %draw
-	#+clim-0.9 :foreground
-	#+clim-1.0 clim:+foreground+
-	#+clim-2 clim:+foreground-ink+
-	#-clim :draw)
-(defvar %erase
-	#+clim-0.9 :background
-	#+clim-1.0 clim:+background+
-	#+clim-2 clim:+background-ink+
-	#-clim :erase)
+(defvar %flip clim:+flipping-ink+)
+(defvar %draw clim:+foreground-ink+)
+(defvar %erase clim:+background-ink+)
 (defvar %alu %draw)
 
-(defun make-color-rgb (red green blue)
-  #-clim (declare (special color:color-screen))
-  #FEATURE-CASE
-  (((not :clim)
-    (when (color:color-exists-p)
-      (scl:send color:color-screen :compute-rgb-alu tv:alu-seta red green blue)))
-   (:clim-2
-    (clim:make-rgb-color red green blue))
-   ((or :clim-1.0 :clim-0.9)
-    (clim:make-color-rgb red green blue))))
-
-(defun color-exists-p ()
-  #-clim (color:color-exists-p)
-  #+clim t)
-
 (defun color-stream-p (stream)
   "Is STREAM capable of rendering color?"
-  #FEATURE-CASE
-  (((not :clim)
-    (and (color:color-exists-p)
-	 ;; error, for example if stream = cold-load-stream
-	 (ignore-errors (color:color-stream-p stream))))
-   (:clim-2
-    (and (clim:extended-output-stream-p stream)
-	 (or (and
-	      (type-specifier-p #-mcclim 'postscript-clim::postscript-stream
-				#+mcclim 'clim-postscript::postscript-stream)
-	      (typep stream
-		     #-mcclim 'postscript-clim::postscript-stream
-		     #+mcclim 'clim-postscript::postscript-stream))
-	     #+ignore		; disallowed
-	     (clim:palette-color-p 
-	      (clim:medium-palette (clim:sheet-medium stream)))
-	     #+ignore		; currently broken & undocumented
-	     (clim:color-stream-p stream)
-	     #-mcclim		; "official" answer.
-	     (clim:palette-color-p
-	      (let ((frame (clim:pane-frame stream)))
-		(if frame
-		    (clim:frame-palette frame)
-		  (clim:port-default-palette
-		   (clim:port (clim:sheet-medium stream))))))
-	     #+mcclim t)))
-   (:clim-0.9
-    (if (typep stream 'ci::encapsulating-stream-mixin)
-	(color-stream-p (slot-value stream 'ci::stream))
-      (let ((port (clim:port stream)))
-	(cond #+Genera
-	      ((typep port 'on-genera::genera-port)
-	       (dolist (inf (tv:sheet-inferiors (on-genera::genera-screen port)))
-		 (ignore-errors
-		  (return (color:color-stream-p inf)))))
-	      #+XLIB
-	      ((typep port 'on-x::x-port)
-	       (> (slot-value port 'on-x::depth) 1))
-	      (t nil)))))
-   ((and :clim-1.0 :clx)
-    (if (typep stream 'clim::clx-window)
-	(slot-value stream 'clim::color-p)
-      t))
-   ((and :clim-1.0 (not mcl))
-    (if (typep stream 'clim::sheet-window-stream)
-	(slot-value stream 'clim::color-p)
-      t))
-   ((and :clim-1.0 mcl)
-    (ccl::screen-color-p ccl::*screen-gdevice*))))
+  (and (clim:extended-output-stream-p stream)
+       (or (and
+            (type-specifier-p 'clim-postscript::postscript-stream)
+            (typep stream 'clim-postscript::postscript-stream))
+           t)))
 
```

Comment:
Sure, and hopefully all this stuff just goes away soon :) I think all of our streams can represent colors.